components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -183,17 +183,30 @@ protected boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) throw
     public boolean validateRequestObject(RequestObject requestObject, OAuth2Parameters oAuth2Parameters)
             throws RequestObjectException {
 
-        boolean isValid = validateClientIdAndResponseType(requestObject, oAuth2Parameters) && checkExpirationTime
-                (requestObject);
-        if (isParamPresent(requestObject, Constants.REQUEST_URI)) {
-            isValid = false;
-        } else if (isParamPresent(requestObject, Constants.REQUEST)) {
-            isValid = false;
-        } else if (requestObject.isSigned()) {
-            isValid = isValidIssuer(requestObject, oAuth2Parameters) && isValidAudience(requestObject,
-                    oAuth2Parameters);
+        // Throws exception if invalid.
+        validateClientIdAndResponseType(requestObject, oAuth2Parameters);
```

Comment:
The default implementation of validateClientIdAndResponseType won't send false but throws exceptions. 

Since validateClientIdAndResponseType is a protected method, we better honor the response of this method. If it returns false, we have to return false.

Reply:
fixed